### PR TITLE
Rearchitect Developer Section

### PR DESCRIPTION
**Pull request by YazzyYaz**
Wanted to split out a lot of repetitive steps into their own pages and sections. Once this is approved and merged, I will make a new PR to link existing and new PRs to those sections after merging `main` into those branches.

### CODE REVIEW

File: docs/developers/environment.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+```sh
+sudo apt install curl tar wget clang pkg-config libssl-dev jq build-essential \
+git make ncdu -y
+```
+
+## Install Golang
+
+Golang will be installed on this machine in order for us to be able to build
+the necessary binaries for running the bridge node. For Golang specifically,
+it’s needed to be able to compile Celestia Application.
```
Comment (evan-forbes) [inline docs/developers/environment.md:30]:
```suggestion
it’s needed to be able to compile the Celestia Application.
```

File: docs/developers/environment.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Setting Up Development Environment
```
Comment (evan-forbes) [inline docs/developers/environment.md:1]: nice, I like separating this out

**Review by evan-forbes — COMMENTED** [review]
nice, only left a few nits

LGTM, but I'll defer to other reviewers

File: docs/developers/celestia-app.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+using the `--help` flag:
+
+```sh
+celestia-appd --help
+```
+
+You should see a similar output (with helpful example commands):
+
+```text
+Stargate CosmosHub App
```
Comment (liamsi) [inline docs/developers/celestia-app.md:34]: I think this shows:
```suggestion
Start celestia app
```

@evan-forbes can confirm.

File: docs/developers/celestia-node.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Celestia Node
+
+This tutorial goes over installing and building Celestia Node. This
+tutorial presumes you completed the steps in setting up your own
+environment [here](../developers/environment).
+
+## Install Celestia Node
+
+Install the Celestia Node binary, which will be used to run the Bridge Node.
```
Comment (liamsi) [inline docs/developers/celestia-node.md:9]: Is there a specific reasons this mentions a bridge node here? the instructions below are independent of the mode the node is ran in later 🤔

Comment (YazzyYaz) [reply docs/developers/celestia-node.md:9]: You're right, I think it crept up from copy-pasta of other page. Will remove it. Good catch!

File: docs/developers/celestia-app.md (hunk 3 above)
Comment (YazzyYaz) [reply docs/developers/celestia-app.md]: I pushed up a commit to fix.

**Review by liamsi — APPROVED** [review]

**Pull request merged**